content_scripts/hud.coffee

```diff
@@ -42,6 +56,28 @@ HUD =
     else
       @tween.fade 0, 150, => @hide true, updateIndicator
 
+  hideFindMode: (data) ->
+    # An element element won't receive a focus event if the search landed on it while we were in the HUD
+    # iframe. To end up with the correct modes active, we create a focus/blur event manually after refocusing
+    # this window.
+    window.focus()
+
+    sel = window.getSelection()
+    focusNode = if not sel.focusNode?
+      null
+    else if sel.focusNode == sel.anchorNode and sel.focusOffset == sel.anchorOffset
+      # The selection either *is* an element, or is inside an opaque element (eg. <input>).
+      sel.focusNode.childNodes[sel.focusOffset]
+    else if sel.focusNode.nodeType != sel.focusNode.ELEMENT_NODE
+      sel.focusNode.parentElement
+    else
+      sel.focusNode
+    document.activeElement?.blur()
+    focusNode?.focus()
```

Comment:
Rather than working out whether we need to focus an input here, might we not instead remember where the selection landed in `execureFind` and restore whatever state we need here?